DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_vlan_increment_mac.py

```diff
@@ -0,0 +1,157 @@
+                                 {"vlan": 33, "untagged": False, "pvid": False},
+                                 {"vlan": 20, "untagged": False, "pvid": False}]},
+
+        {"port": 3, "settings": [{"vlan": 5, "untagged": False, "pvid": True},
+                                 {"vlan": 33, "untagged": False, "pvid": False},
+                                 {"vlan": 20, "untagged": False, "pvid": False}]})
+
+    await configure_vlan_setup(device, port_map, dut_ports)
+
+    dev_groups = tgen_utils_dev_groups_from_config(
+        [{"ixp": tg_ports[0], "ip": "100.1.1.2", "gw": "100.1.1.6", "plen": 24, },
+         {"ixp": tg_ports[1], "ip": "100.1.1.3", "gw": "100.1.1.6", "plen": 24, },
+         {"ixp": tg_ports[2], "ip": "100.1.1.4", "gw": "100.1.1.6", "plen": 24, },
+         {"ixp": tg_ports[3], "ip": "100.1.1.5", "gw": "100.1.1.6", "plen": 24, }])
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, dut_ports, dev_groups)
+
+    tx_ports = dev_groups[tg_ports[0]][0]["name"]
+    rx_ports = dev_groups[tg_ports[1]][0]["name"]
+
+    # 4. Set up streams with incremented MAC address (16000), pps 15000 and vlans 5, 33, 20
+    for vlan in packet_vids:
+        streams = {f"traffic with VLAN ID: {vlan}": {
+            "type": "raw",
+            "protocol": "802.1Q",
+            "ip_source": tx_ports,
+            "ip_destination": rx_ports,
+            "srcMac": {"type": "increment",
+                       "start": "02:00:00:00:ff:00",
+                       "step": "00:00:00:00:10:00",
+                       "count": mac_count},
+            "dstMac": "ff:ff:ff:ff:ff:ff",
+            "rate": 15000,
+            "vlanID": vlan
+        }}
+
+        await tgen_utils_setup_streams(tgen_dev, config_file_name=None, streams=streams)
+
+        # 5. Send traffic
+        await tgen_utils_start_traffic(tgen_dev)
+        await asyncio.sleep(15)
+        await tgen_utils_stop_traffic(tgen_dev)
+
+        # 6. Verify no packet loss nor packet leak occurred and all transmitted traffic received
+        stats = await tgen_utils_get_traffic_stats(tgen_dev, "Flow Statistics")
+        ti_to_rx_port_map = get_traffic_port_vlan_mapping(streams, port_map, tg_ports)
+        for row in stats.Rows:
+            if row["Rx Port"] in ti_to_rx_port_map[row["Traffic Item"]]:
+                assert tgen_utils_get_loss(row) == 0.000, \
+                    f'Traffic leak for traffic item: {row["Traffic Item"]} on port {row["Rx Port"]}'
+            else:
+                assert tgen_utils_get_loss(row) == 100.000, \
+                    f'No traffic for traffic item : {row["Traffic Item"]} on port {row["Rx Port"]}'
+
+        # 7. Verify all MAC entries were learnt or re-learnt on the VLAN
+        _, number_of_macs = await dent_devices[0].run_cmd(
+            f"bridge fdb show dev {dut_ports[0]} | grep 'vlan {vlan}' | wc -l")
+        assert int(number_of_macs.strip()) > (mac_count * tolerance), f"Fail learning all macs with vlan: {vlan}\
```

Comment:
consider moving your error message to a new line:
```suggestion
        assert int(number_of_macs.strip()) > (mac_count * tolerance), \
            f"Fail learning all macs with vlan: {vlan} ...
```